pkg/solana/txm/utils.go

```diff
@@ -51,6 +51,35 @@ func (s TxState) String() string {
 	}
 }
 
+type regressionType int
+
+const (
+	FromConfirmed regressionType = iota
+	FromProcessed
+)
+
+// isStatusRegression checks if the current status is a regression compared to the previous status:
+// - Confirmed -> Processed, Broadcasted: should not regress
+// - Processed -> Broadcasted: should not regress
+// Returns true if a regression is detected, indicating a possible re-org.
+func isStatusRegression(previous, current TxState) (regressionType, bool) {
+	switch previous {
+	case Confirmed:
+		// Confirmed transactions should not regress to Processed or Broadcasted.
+		if current != Confirmed && current != Finalized && current != Errored {
```

Comment:
Couldn't we simplify this if by checking for Broadcasted or Processed?
```suggestion
		if current == Broadcasted || current == Processed {
```